### PR TITLE
Fuse <> Gnosis bridge

### PR DESCRIPTION
**Pull request by enok1111**

### CODE REVIEW

File: constants/uniswapRouters.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"fuse": "0xF4d73326C13a4Fc5FD7A064217e12780e9Bd62c3",
```
Comment (leonprou) [inline constants/uniswapRouters.json:2]: @en0c-026 any idea how uniswap can be used?

File: deploy/NativeOFT.js [hunk 2]
```diff
@@ -2,23 +2,19 @@ const LZ_ENDPOINTS = require("../constants/layerzeroEndpoints.json")
 const NATIVE_OFT_ARGS = require("../constants/nativeOftArgs.json")
 
 module.exports = async function ({ deployments, getNamedAccounts }) {
-    const { deploy } = deployments
-    const { deployer } = await getNamedAccounts()
-    console.log(`>>> your address: ${deployer}`)
+  const { deploy } = deployments
```
Comment (leonprou) [inline deploy/NativeOFT.js:5]: any changes here? If not please restore

File: deploy/SwappableBridge.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const ROUTERS = require("../constants/uniswapRoutes.json")
+const ROUTERS = require("../constants/uniswapRouters.json")
```
Comment (leonprou) [inline deploy/SwappableBridge.js:1]: Hi @en0c-026 , why this was changed?

File: tasks/verifyContract.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
+const FileSystem = require("fs");
+const BLOCK_EXPLORER_API_URL = require("../constants/blockExplorerApi.json")
```
Comment (leonprou) [inline tasks/verifyContract.js:2]: you can use hardhat-etherscan plugin to verify the contracts (I know it's counterintuitive)

File: contracts/SwappableBridge.sol [hunk 5]
```diff
@@ -6,27 +6,26 @@ import "@layerzerolabs/solidity-examples/contracts/token/oft/IOFTCore.sol";
...
     }
 
-    function swapAndBridge(uint amountIn, uint amountOutMin, uint16 dstChainId, address to, address payable refundAddress, address zroPaymentAddress, bytes calldata adapterParams) external payable {
+    function swapAndBridge(address _oft, uint amountIn, uint amountOutMin, uint16 dstChainId, address to, address payable refundAddress, address zroPaymentAddress, bytes calldata adapterParams) external payable {
+        require(_oft != address(0), "SwappableBridge: invalid OFT address");
         require(to != address(0), "SwappableBridge: invalid to address");
         require(msg.value >= amountIn, "SwappableBridge: not enough value sent");
-
+        
+        IOFTCore oft = IOFTCore(_oft);
```
Comment (leonprou) [inline contracts/SwappableBridge.sol:25]: Hi @en0c-026 can you explain the change?

**Review by leonprou — COMMENTED** [review]
please see my questions

File: deploy/SwappableBridge.js (hunk 3 above)
Comment (enok1111) [reply deploy/SwappableBridge.js:1]: In my opinion the file is wrongly named, I changed it to make it clearer, they are the addresses of the Uniswap Routers in each network.

File: tasks/verifyContract.js (hunk 4 above)
Comment (enok1111) [reply tasks/verifyContract.js:2]: I would prefer to leave it like this it works perfectly with a single command the contracts are verified

File: contracts/SwappableBridge.sol (hunk 5 above)
Comment (enok1111) [reply contracts/SwappableBridge.sol:25]: This change is to be able to use the same SwappableBridge contract for more than one chain in Fuse, so before the contract was configured with the token when deploying.
Now that is done by passing the address of the OFT token to the swapAndBridge function, so you can use it for gnosis as well as metis.

File: deploy/NativeOFT.js (hunk 2 above)
Comment (enok1111) [reply deploy/NativeOFT.js:5]: Yes, the tabs in the code were formatted correctly

File: contracts/SwappableBridge.sol (hunk 5 above)
Comment (leonprou) [reply contracts/SwappableBridge.sol:25]: OK we should be cautious with changes in the smart contracts.
worth to check this with the LZ team
 @MayankMittal1  FYI

Comment (MayankMittal1) [reply contracts/SwappableBridge.sol:25]: @en0c-026 is this recommended and approved for use by LayerZero?

**Pull request closed without merging**